src/opossum_lib/core/services/merge_opossums.py

```diff
@@ -0,0 +1,233 @@
+# SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import datetime
+import logging
+import uuid
+from collections import OrderedDict
+from collections.abc import Iterable
+from pathlib import PurePath
+from typing import cast
+
+from opossum_lib.core.entities.base_url_for_sources import BaseUrlsForSources
+from opossum_lib.core.entities.external_attribution_source import (
+    ExternalAttributionSource,
+)
+from opossum_lib.core.entities.frequent_license import FrequentLicense
+from opossum_lib.core.entities.metadata import Metadata
+from opossum_lib.core.entities.opossum import (
+    Opossum,
+    ScanResults,
+)
+from opossum_lib.core.entities.opossum_package import OpossumPackage
+from opossum_lib.core.entities.resource import Resource
+from opossum_lib.shared.entities.opossum_output_file_model import (
+    Metadata as OutputMetadata,
+)
+from opossum_lib.shared.entities.opossum_output_file_model import OpossumOutputFileModel
+
+
+def merge_opossums(opossums: list[Opossum]) -> Opossum:
+    if len(opossums) < 2:
+        raise RuntimeError(
+            f"You need to provide at least 2 opossums for merging. Got: {len(opossums)}"
+        )
+    scan_results = _merge_scan_results(opossums)
+    review_results = _handle_review_results(opossums, scan_results)
+    return Opossum(
+        scan_results=scan_results,
+        review_results=review_results,
+    )
+
+
+def _handle_review_results(
+    opossums: list[Opossum], scan_results: ScanResults
+) -> OpossumOutputFileModel | None:
+    review_results = _extract_review_results(opossums)
+    if len(review_results) == 0:
+        return None
+    elif len(review_results) > 1:
+        raise RuntimeError(
+            "More than one .opossum input contains review results. "
+            + f"This is currently unsupported. Got: {len(review_results)}"
+        )
+    else:
+        new_metadata = OutputMetadata(
+            project_id=scan_results.metadata.project_id,
+            file_creation_date=scan_results.metadata.file_creation_date,
+            input_file_md5_checksum=None,
+        )
+        return review_results[0].model_copy(update={"metadata": new_metadata})
+
+
+def _extract_review_results(opossums: list[Opossum]) -> list[OpossumOutputFileModel]:
+    review_results = [
+        opossum.review_results
+        for opossum in filter(_has_nonempty_review_result, opossums)
+    ]
+
+    return cast(list[OpossumOutputFileModel], review_results)
+
+
+def _has_nonempty_review_result(opossum: Opossum) -> bool:
+    return bool(
+        opossum.review_results
+        and (
+            opossum.review_results.manual_attributions
+            or opossum.review_results.resolved_external_attributions
+            or opossum.review_results.resources_to_attributions
+        )
+    )
```

Comment:
there are multiple issues here:
1. it's not common to use functional programming in python and in this case there is also no need to.
2. by using `filter` in combination with a filter function you're masking the real type of `opossum.review_results` which means you need to use a cast in line 70. but this, too, is not necessary.

the code becomes simpler when you filter the way you usually would inside a comprehension:
```suggestion
def _extract_review_results(opossums: list[Opossum]) -> list[OpossumOutputFileModel]:
    return [
        opossum.review_results
        for opossum in opossums
        if opossum.review_results
        and (
            opossum.review_results.manual_attributions
            or opossum.review_results.resolved_external_attributions
            or opossum.review_results.resources_to_attributions
        )
    ]
```